---
title: "C6310 | Документы Microsoft"
ms.custom: 
ms.date: 11/04/2016
ms.reviewer: 
ms.suite: 
ms.technology:
- vs-ide-code-analysis
ms.tgt_pltfrm: 
ms.topic: article
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
caps.latest.revision: 
author: mikeblome
ms.author: mblome
manager: ghogen
ms.workload:
- multiple
ms.openlocfilehash: 1ae678ff072e317f33e7873484131d305d991133
ms.sourcegitcommit: 32f1a690fc445f9586d53698fc82c7debd784eeb
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 12/22/2017
---
# <a name="c6310"></a>C6310
Предупреждение C6310: Недопустимая константа в фильтре исключений может стать причиной непредвиденного поведения  
  
 Это сообщение означает, что в выражении фильтра структурированного обработчика исключений обнаружена недопустимая константа. Ниже перечислены константы, определенные для использования в выражении фильтра структурированного обработчика исключений  
  
-   `EXCEPTION_CONTINUE_EXECUTION`  
  
-   `EXCEPTION_CONTINUE_SEARCH`  
  
-   `EXCEPTION_EXECUTE_HANDLER`  
  
 Эти значения определяются в excpt.h файл заголовка среды выполнения.  
  
 Использование константы, не перечисленные в списке может привести к непредвиденному поведению.  
  
## <a name="example"></a>Пример  
 Это предупреждение возникает в следующем коде:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Чтобы устранить это предупреждение, используйте следующий код:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```